djconnectwise/sync.py

```diff
@@ -1439,6 +1439,53 @@ def get_page(self, *args, **kwargs):
         return self.client.get_project_types(*args, **kwargs)
 
 
+class ProjectTeamMemberSynchronizer(Synchronizer):
+    client_class = api.ProjectAPIClient
+    model_class = models.ProjectTeamMember
+
+    related_meta = {
+        'member': (models.Member, 'member'),
+        'workRole': (models.WorkRole, 'work_role')
+    }
+
+    def _assign_field_data(self, instance, json_data):
+        instance.id = json_data['id']
+        instance.start_date = json_data.get('startDate')
+        instance.end_date = json_data.get('endDate')
+
+        try:
+            project_id = json_data['projectId']
+            related_project = models.Project.objects.get(pk=project_id)
+            setattr(instance, 'project', related_project)
+        except KeyError:
+            raise InvalidObjectException(
+                'Project team member {} has no projectId key to find its '
+                'target - skipping.'.format(instance.id)
+            )
+        except ObjectDoesNotExist as e:
+            raise InvalidObjectException(
+                'Project team member {} has a projectId that does not exist.'
+                ' ObjectDoesNotExist Exception: {}'.format(instance.id, e)
+            )
+
+        self.set_relations(instance, json_data)
+        return instance
+
+    def client_call(self, project_id, *args, **kwargs):
+        return self.client.get_project_team_members(
+            project_id, *args, **kwargs)
+
+    def get_page(self, *args, **kwargs):
+        records = []
+        project_qs = models.Project.objects.filter(
+            status__closed_flag=False).order_by(self.lookup_key)
+
+        for project_id in project_qs.values_list('id', flat=True):
+            records += self.client_call(project_id, *args, **kwargs)
```

Comment:
I see this class doesn't inherit from `BatchConditionMixin`, so I think that means if you were to try and sync team member records with an instance that has sync'd around 400+ open projects it will probably blow up. Now I know that isn't typical but I would put dollars to donuts that we will encounter a few clients with terribly messy instances, and it will cause lots of problems with setting up and syncing and we will have to stop everything to fix it before continuing. With `BatchConditionMixin` it will be able to find the optimal url size with the given project ids, so it doesn't make one too long and explode.

Reply:
Since the endpoint for `ProjectTeamMember` was quite similar to the endpoint for `ServiceNote` I used `ServiceNoteSynchronizer` as an example when implementing `ProjectTeamMemberSynchronizer`. I guess I didn't consider batching too closely since the service note synchronizer doesn't use it. 

I'll look into this a bit but if we are calling `/v4_6_release/apis/3.0/project/projects/{ project_id }/teamMembers` do we still need to consider batching?

Reply:
No sorry I must have misread, I thought we were requesting them in batches like Tickets, i.e. "statusId+In=(1,2,3,4,5,6... etc)". If the requests are being made like service notes then this should be fine.

Reply:
Ok, great. 

I added the changes for admin search fields. Ready for another look.